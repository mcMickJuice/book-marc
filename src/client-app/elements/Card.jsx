import React, {PropTypes as T} from 'react'

const Card = ({children, className}) => {
    return <div className={`bm-card ${className}`}>
        {children}
    </div>
}

Card.propTypes = {
    children: T.any.isRequired,
    className: T.string
}

export default Card