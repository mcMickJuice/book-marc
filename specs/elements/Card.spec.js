import React from 'react'
import Card from 'elements/Card'
import renderer from 'react-test-renderer'

//first react test with jest so its simple
//dont judge me
test('className passed in should be in classList', () => {
    const component = renderer.create(
        <Card className="my-class">
            Hello
        </Card>
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('children passed in should be rendered in container', () => {
    const component = renderer.create(
        <Card className="super-awesome-class">
            This is the text in the Card
        </Card>
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})