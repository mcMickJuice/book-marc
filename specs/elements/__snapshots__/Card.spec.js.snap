// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`children passed in should be rendered in container 1`] = `
<div
  className="bm-card super-awesome-class"
>
  This is the text in the Card
</div>
`;

exports[`className passed in should be in classList 1`] = `
<div
  className="bm-card my-class"
>
  Hello
</div>
`;
